tools/rdm/TestDefinitions.py

```diff
@@ -6473,14 +6437,150 @@ class GetListInterfacesWithData(TestMixins.GetWithDataMixin,
+  """GET the IPv4 default route."""
+  # TODO(Peter): Check interface identifier is a valid interface
+  CATEGORY = TestCategory.IP_DNS_CONFIGURATION
+  PID = 'IPV4_DEFAULT_ROUTE'
+  EXPECTED_FIELD = ['ipv4_address', 'interface_identifier']
+  # TODO(Peter): Is this required?
+  #ALLOWED_NACKS = [RDMNack.NR_HARDWARE_FAULT]
+
+class GetIPv4DefaultRouteWithData(TestMixins.GetWithDataMixin,
+                                  OptionalParameterTestFixture):
+  """Get IPv4 default route with param data."""
+  CATEGORY = TestCategory.ERROR_CONDITIONS
+  PID = 'IPV4_DEFAULT_ROUTE'
+
+# TODO(Peter): Need to restrict these somehow so we don't saw off the branch
+#class SetIPv4DefaultRoute(TestMixins.UnsupportedSetMixin,
+#                          OptionalParameterTestFixture):
+#  """Attempt to SET the IPv4 default route with no data."""
+#  CATEGORY = TestCategory.ERROR_CONDITIONS
+#  PID = 'IPV4_DEFAULT_ROUTE'
+#
+#
+#class SetIPv4DefaultRouteWithData(TestMixins.UnsupportedSetMixin,
+#                                  OptionalParameterTestFixture):
+#  """SET the IPv4 default route with data."""
+#  CATEGORY = TestCategory.ERROR_CONDITIONS
+#  PID = 'IPV4_DEFAULT_ROUTE'
+#  DATA = 'FOOBAR'
+
+class AllSubDevicesGetIPv4DefaultRoute(TestMixins.AllSubDevicesGetMixin,
+                                       OptionalParameterTestFixture):
+  """Send a Get IPV4_DEFAULT_ROUTE to ALL_SUB_DEVICES."""
+  CATEGORY = TestCategory.SUB_DEVICES
+  PID = 'IPV4_DEFAULT_ROUTE'
+
+# Interface label
+#------------------------------------------------------------------------------
+class GetInterfaceLabels(TestMixins.GetSettingDescriptionsMixinList,
+                         OptionalParameterTestFixture):
+  """Get the interface labels for all defined interfaces."""
+  CATEGORY = TestCategory.IP_DNS_CONFIGURATION
+  PID = 'INTERFACE_LABEL'
+  REQUIRES = ['interface_list']
+  EXPECTED_FIELD = 'interface_identifier'
+  DESCRIPTION_FIELD = 'interface_label'
+
+class GetInterfaceLabelWithNoData(TestMixins.GetWithNoDataMixin,
+                                  OptionalParameterTestFixture):
+  """Get the interface label with no interface id specified."""
+  CATEGORY = TestCategory.ERROR_CONDITIONS
+  PID = 'INTERFACE_LABEL'
+
+
+class GetInterfaceLabelWithTooMuchData(OptionalParameterTestFixture):
+  """Get the interface label with more than 4 bytes of data."""
+  CATEGORY = TestCategory.ERROR_CONDITIONS
+  PID = 'INTERFACE_LABEL'
+
   def Test(self):
-    self.AddExpectedResults(TestMixins.UnsupportedSetNacks(self.pid))
-    self.SendRawSet(ROOT_DEVICE, self.pid)
+    self.AddIfGetSupported(self.NackGetResult(RDMNack.NR_FORMAT_ERROR))
+    self.SendRawGet(ROOT_DEVICE, self.pid, 'foobar')
+
+class GetZeroInterfaceLabel(TestMixins.GetZeroByteMixin,
+                            OptionalParameterTestFixture):
+  """GET INTERFACE_LABEL for interface 0."""
+  CATEGORY = TestCategory.ERROR_CONDITIONS
+  PID = 'INTERFACE_LABEL'
+  DATA = struct.pack('!L', 0)
```

Comment:
See other comment - don't change the width from 8 to 32, use a separate mixin instead.